common4j/src/main/com/microsoft/identity/common/java/nativeauth/controllers/results/INativeAuthCommandResult.kt

```diff
@@ -33,13 +33,15 @@ interface INativeAuthCommandResult : ILoggable {
     val correlationId: String
     data class Redirect(
         override val correlationId: String,
+        val redirectReason: String
     ) : Error(error = BROWSER_REQUIRED_ERROR, errorDescription = BROWSER_REQUIRED_ERROR_DESCRIPTION, correlationId = correlationId),
```

Comment:
What if you make this field optional and you do something like this:
```suggestion
    ) : Error(error = BROWSER_REQUIRED_ERROR, errorDescription = redirectReason :? BROWSER_REQUIRED_ERROR_DESCRIPTION, correlationId = correlationId),
```
So, you use redirectReason when available and "BROWSER_REQUIRED_ERROR_DESCRIPTION" as default error message

Reply:
That's a good suggestion. Combined with the discussion above, what do you think the final version of
```
data class Redirect(
        override val correlationId: String,
        val redirectReason: String
    ) : Error(error = BROWSER_REQUIRED_ERROR, errorDescription = redirectReason.ifEmpty { BROWSER_REQUIRED_ERROR_DESCRIPTION }, correlationId = correlationId),
```

Reply:
I'm ok, as long as we use "BROWSER_REQUIRED_ERROR_DESCRIPTION" whenever we don't have a "redirectReason"